utils/firestore.js

```diff
@@ -1041,10 +1041,43 @@ const updateSpecimen = async (id, data) => {
     await db.collection('biospecimen').doc(docId).update(data);
 }
 
+//TODO: remove this function after Aug 2023 push. Verify addBoxAndUpdateSiteDetails() is live and working as expected.
 const addBox = async (data) => {
     await db.collection('boxes').add(data);
 }
 
+// atomically create a new box in the 'boxes' collection and update the 'siteDetails' doc with the most recent boxId as a numeric value
+const addBoxAndUpdateSiteDetails = async (data) => {
+    try {
+        const boxDocRef = db.collection('boxes').doc();
+        const siteDetailsDocRef = db.collection('siteDetails').doc(data['siteDetailsDocRef']);
+        delete data['siteDetailsDocRef'];
+
+        if (!siteDetailsDocRef) {
+            throw new Error("siteDetailsDocRef is not provided in the data object.");
+        }
+
+        const boxIdString = data['132929440'];
```

Comment:
might suggest added a Concept ID constant

Reply:
Sure, added `shippingBoxId` to `fieldToConceptIdMapping.js`.